### PR TITLE
(SIMP-7032) Add set_travis_env_vars.rb + Tasks

### PR DESCRIPTION
**Pull request by op-ct**
This commit adds the  **set_travis_env_vars.rb** script, which sets or
deletes an organization's secrets (like API tokens) across multiple
Travis CI projects.

It also introduces Puppet Tasks support for the repository,
and adds the task `releng_tasks::set_travis_env_vars`

SIMP-7070 #close
SIMP-7069 #close
SIMP-7071 #close
SIMP-7032 #close

### CODE REVIEW

File: Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb [hunk 1]
```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env ruby
+#
+# Set/delete a common env var across all Travis CI repos in an organization
+#
+# * Requires a Travis CI token set in environment variable `TRAVIS_TOKEN`
+# * Uses Travis CI API v3 (https://developer.travis-ci.org)
+#
+# @author Name Chris Tessmer <chris.tessmer@onyxpoint.com>
+# @license https://apache.org/licenses/LICENSE-2.0
```
Comment (lnemsick-simp) [inline Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb:9]: Isn't the LICENSE for this GitHub project sufficient?

File: Boltdir/site-modules/releng_tasks/files/http_request.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+require 'net/http'
+require 'uri'
+require 'json'
+require 'openssl'
+
+# A fit-for-most-purposes, MRI-compatible HTTP/S swiss army knife method
```
Comment (lnemsick-simp) [inline Boltdir/site-modules/releng_tasks/files/http_request.rb:6]: MRI = Ruby MRI?

File: Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      Usage:
+
+        TRAVIS_TOKEN=<TOKEN> #{File.basename($PROGRAM_NAME)} [options] ORG VARIABLE [VALUE]
+
+      Note:
+
+        The environment variable TRAVIS_TOKEN must contain your Travis CI token
+    HELP_MSG
+
+    opts.separator ''
```
Comment (lnemsick-simp) [inline Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb:178]: Why isn't the filter exposed?  For the simp org, wouldn't we want the filter for forked projects we don't own?

File: Boltdir/site-modules/releng_tasks/files/http_request.rb (hunk 2 above)
Comment (trevor-vaughan) [reply Boltdir/site-modules/releng_tasks/files/http_request.rb:6]: @lnemsick-simp Yeah, that's standard shorthand.

File: Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb (hunk 3 above)
Comment (trevor-vaughan) [reply Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb:178]: @lnemsick-simp Edit: `repo_filter` is exposed through the bolt task params and the initialize function.

Comment (lnemsick-simp) [reply Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb:178]: @trevor-vaughan  If you execute this file as is, there is no way to set `repo_filter` from a command line option.  The options parser does not support setting it.  This is distinct from `tasks/set_travis_env_vars.rb`, which allows you to set it with options read in from a JSON file.

Comment (trevor-vaughan) [reply Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb:178]: @op-ct Did you want to do anything with this?

**Review by trevor-vaughan — APPROVED** [review]

**Pull request merged**